pkg/repositories/database.go

```diff
@@ -94,14 +104,16 @@ func GetDB(ctx context.Context, dbConfig *runtimeInterfaces.DbConfig, logConfig
 			ExtraOptions: dbConfig.DeprecatedExtraOptions,
 			Debug:        dbConfig.DeprecatedDebug,
 		}
-		dialector = postgres.Open(getPostgresDsn(ctx, pgConfig))
+
 	default:
 		panic(fmt.Sprintf("Unrecognized database config %v", dbConfig))
 	}
-	gormDb, err := gorm.Open(dialector, &gorm.Config{
+
+	gormConfig := &gorm.Config{
```

Comment:
can we please initialize the dialector in the switch statements? (the whole purpose of the previous refactor so we could easily add new gorm-supported database types)

Reply:
makes sense. Have fixed this.

Wanted o fix that part but i think woud be good if we can do switch based on some provider attribute in the main dbConfig that way each switch case is working on deprecated, postgres, mysql ... other providers.

Let me know what you think